src/kakarot/library.cairo

```diff
@@ -115,4 +179,41 @@ namespace Kakarot {
         native_token_address.write(native_token_address_);
         return ();
     }
+
+    // @notice Deploy the starknetcontract holding the evm code
+    // @param bytes: byte code stored in the new contract
+    // @return evm_contract_address: address that is mapped to the actual new contract address
+    @external
+    func deploy_contract{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        bytes_len: felt, bytes: felt*
+    ) -> (evm_contract_address: felt, starknet_contract_address: felt) {
+        alloc_locals;
+        let (current_salt) = salt.read();
+        let (class_hash) = evm_contract_class_hash.read();
+
+        let (local calldata: felt*) = alloc();
+        let (kakarot_address) = get_contract_address();
+
+        assert [calldata] = kakarot_address;
+        assert [calldata + 1] = bytes_len;
+        memcpy(dst=calldata + 2, src=bytes, len=bytes_len);
+
+        let (contract_address) = deploy(
+            class_hash=class_hash,
+            contract_address_salt=current_salt,
+            constructor_calldata_size=bytes_len + 2,
+            constructor_calldata=calldata,
+            deploy_from_zero=FALSE,
+        );
+        salt.write(value=current_salt + 1);
+        // Generate EVM_contract address from the new cairo contract
+        // TEMPORARY SOLUTION FOR HACK-LISBON !!!
```

Comment:
🥹

Reply:
actually I don't know how this address should be computed, do you?

Reply:
not entirely sure! I guess it can work for now but we should flag it as a TODO then

Reply:
and maybe add `assert address = 9` to make sure?!